depdive/src/main.rs

```diff
@@ -1,43 +1,47 @@
 use anyhow::Result;
-use depdive::{
-    DependencyGraphAnalyzer, TabledCrateSourceDiffReport, TabledCratesioReport, TabledGitHubReport,
-};
-use guppy::MetadataCommand;
-use tabled::{Style, Table};
+use depdive::UpdateAnalyzer;
+use structopt::StructOpt;
 
-fn main() -> Result<()> {
-    let graph = MetadataCommand::new().build_graph()?;
-
-    let depdive_report = DependencyGraphAnalyzer.analyze_dep_graph(&graph)?;
+#[derive(Debug, StructOpt)]
+#[structopt(about = "Rust dependency analysis")]
+struct Args {
+    #[structopt(subcommand)]
+    cmd: Command,
+}
 
-    let table: Vec<TabledCratesioReport> = depdive_report
-        .crate_stats
-        .iter()
-        .map(|r| r.tabled_cratesio_report.clone())
-        .collect();
-    let table = Table::new(table).with(Style::github_markdown()).to_string();
-    println!("{}", table);
+#[derive(Debug, StructOpt)]
+enum Command {
+    #[structopt(name = "update-review")]
+    UpdateReview {
+        #[structopt(subcommand)]
+        cmd: UpdateReviewCommand,
+    },
+}
 
-    let table: Vec<TabledGitHubReport> = depdive_report
-        .crate_stats
-        .iter()
-        .map(|r| r.tabled_github_report.clone())
-        .collect();
-    let table = Table::new(table).with(Style::github_markdown()).to_string();
-    println!("{}", table);
+#[derive(Debug, StructOpt)]
+enum UpdateReviewCommand {
+    #[structopt(name = "paths")]
+    Paths { prior: String, post: String },
 
-    let table: Vec<TabledCrateSourceDiffReport> = depdive_report
-        .crate_stats
-        .iter()
-        .map(|r| r.tabled_crate_source_diff_report.clone())
-        .collect();
-    let table = Table::new(table).with(Style::github_markdown()).to_string();
-    println!("{}", table);
+    #[structopt(name = "commits")]
+    Commits {
+        path: String,
+        prior: String,
+        post: String,
```

Comment:
might be nice to add some doc comments to these cmdline arguments to better understand what they are